### ISSUE TITLE
[Question] Playable on Linux?

### ISSUE THREAD

**Issue by Mumrik93** [aa2g/AA2Unlimited#267]
Is there some way to play this game on Linux? Used to play it a lot a few years ago but has migrated to Linux since then and just recently re-discovered AA2. I saw that AA2 is on Lutris, but it doesn't have an install script, sadly enough.

https://lutris.net/games/artificial-academy-2/

**Comment by geneishouko** [comment]
Yeah, it just works with AA2Unlimited and _wined3d_ or _dxvk_, which are included in the package. For an easy to install and ready to play experience you can get AA2Mini. Note that AA2Mini got a big recent update that I haven't checked if it works with Linux yet.

**Comment by Mumrik93** [comment]
This might be a stupid question, but how did make the game itself work on Linux? No matter how I try I'm unable to make the game run, I take it the game only runs with your mod installed? Since it comes packed with wined3d and/or dxvk? Or are there some special settings or packages that need to be installed that I/we should know of?

**Comment by geneishouko** [comment]
The game has always worked since release day, only that it doesn't look good without our fixes. Since you didn't specify what's going on on your side, what worked and what did not, what you did, what was the exact outcome, I can't help you without that. So it was a stupid question, but you can do better.

**Comment by Mumrik93** [comment]
I've tried two times to get AA2 to play on Linux, the first time I wasn't even able to install the game, never managed to get past install, I never knew what the reason was. A year later I tried again, then I made sure to install visual c++ and direct x. I managed somehow to get past install, but then I instead encountered locale-issues, which I did not manage to correct the same way I've done on Windows 10 previously.
Not sure if it helps. Hope it does!

**Comment by geneishouko** [comment]
Wait a minute. Did you actually get the game to install and run? Because you now make it sound like that and that's an entirely different question. Locale issues are fixed by running both the installer and the game with a LANG=ja_JP.utf8 environment variable.

You may want to try a preinstall like AA2Mini here
https://tsukiyo.me/AAA/AA2MiniPPX.xml
I don't know if this pre-install works on Linux but it's worth to try

Or here
https://tsukiyo.me/AAA/AA2Mini.xml
In case the previous one doesn't work. This one is confirmed to work on Linux but it's prone to crash

**Comment by Mumrik93** [comment]
Oh no, The game did not run, it got (unreadable) error messages whenever i attempted to start. I had help back on windows 10 when i installed and i received similar errors back then, and then they said it was a locale issue and the game had Not been installed correctly, that's why i decided to give up here on Linux (on windows I reinstalled) since I'd already set up the locale the same way Id done on Windows and it still didn't work.
I'll give the pre-install a try and get back to you with how it goes! Thanks a lot!

**Comment by Mumrik93** [comment]
Hello again!

I've been trying out the pre-packaed version, but sadly I'm getting errors, they are all defined as "Font config error" and "Font config warning", after said errors the process simply ends, the game itself however never starts :( I got the same result from both the pre-packed versions.

**Comment by geneishouko** [comment]
It's long due time for you to post screenshots and information on your system

**Comment by Mumrik93** [comment]
Yes I guess it is~

OS: Linux Mint 20 Cinnamon
Kernel: 5.4.0-56-generic
CPU: AMD Ryzen 5 1600X Six-Core Processor × 6
GPU: GeForce GTX 1080 Ti (Drives: nvidia-driver-450)

The first was me attempting to run AA2 with Lutris (latest Wine Runner) the second was also with Lutris (but with Proton instead) The third is still running Proton but I disabled VKD3D, just to see if there where any difference. In all three cases DXVK where enabled.

**Comment by geneishouko** [comment]
We need console logs, the wine error messages, if any.
You should also try the regular wine that comes with your distro.

**Comment by geneishouko** [comment]
That system and OS should run the game with no issues

**Comment by Mumrik93** [comment]
Linux Mint does not come pre-packaged with Wine though, but I have installed it.
But! I have just found a way to start and play the game.. I don't know why this is the only way I can get it to work so far, but here it is.

1.) Open up 'Uninstall Wine Software'
2.) Chose 'Install' from the wine menu.
3.) Chose the the AA2Play

This is for some reason the only way I can get it to start, nothing happens if I try to start the game the conventional way trough Nemo, such as double-clicking the .exe or choosing wine in the 'chose-software'-list to open it, then it just loads for a short time... and then nothing..

**Comment by Mumrik93** [comment]
So I've been playing the game a little now and the second pre-package you recommended (the older one) Is playable but it has some bugs. Most letters are replaced by squares, making dialogue and character names impossible to read, and the game is sadly lagging quite a lot. I have not suffered any crashes thou!

The Editor also work and I've not suffered any problems with that one at all, it works flawlessly!

**Comment by geneishouko** [comment]
> Most letters are replaced by squares, making dialogue and character names impossible to read

This is caused by missing Japanese fonts or by playing in the wrong locale. See the environment variable in a previous comment.

> the game is sadly lagging quite a lot

This might be caused by using wined3d and the d3dx9_42 dll override. If you use the wined3d fix, you must not use d3d9 dll overrides. But if you use the dxvk fix, you must use the d3dx9_42 dll override.

**Comment by Mumrik93** [comment]
I do use DXVK so I've turned on the d3d9 override, and it does lag less, but it still lags a little bit, but it's playable! The only negative is the load-time whenever there's an interaction is still a bit long, it takes about 5-10 seconds for the dialogue to pop up.
I've added the "LANG=ja_JP.utf8 environment variable" to activate at boot, but all letters still turn up as squares. Is there a special way to add/turn-on this variable that you'd recommend?

**Comment by geneishouko** [comment]
I just set the variable in the launcher shortcut by prepending `env LANG=ja_JP.utf8` to the command. Your problem might be missing fonts, though. Try copying the Microsoft Japanese fonts in your system or install fakejapanese with winetricks.
